indexer.py

```diff
@@ -42,34 +42,50 @@
 # Path to markdown files
 SOURCE_PATH = "/home/indexer/gitcache"
 
-def clone_repos(repo_url, branch):
+
+def clone_repo(repo_url, branch, target_path):
     """
-    Create a local clone of the docs repository defined by repo_url
-    using the given branch plus all the external repositories
-    referenced by the docs repo in the src/external-repositories.txt file
+    Create a shallow clone of a git repository using a certain branch/tag in
+    a given target folder. If the target folder exists, it will be removed
+    first and then created again.
     """
     logging.info("Cloning git repository %s, branch '%s'" % (repo_url, branch))
 
     # repo name from URL
     (reponame, _) = os.path.basename(repo_url).split(".")
 
-    main_path = SOURCE_PATH + os.sep + reponame
+    if os.path.exists(target_path):
+        shutil.rmtree(target_path)
 
-    if os.path.exists(main_path):
-        shutil.rmtree(main_path)
+    os.makedirs(target_path, exist_ok=True)
 
-    os.makedirs(main_path)
     cmd = ["git", "clone", "-q",
            "--depth", "1",
            "-b", branch,
-           repo_url, main_path]
+           repo_url, target_path]
     returncode = call(cmd)
 
     # check success
     if returncode > 0:
+        return False
+    return True
```

Comment:
```suggestion
    return returncode <= 0
```

Reply:
I find my three lines easier to understand. But thanks :)